packages/stencil-library/src/components/button/button.tsx

```diff
@@ -25,57 +21,87 @@ export class Button {
+                    return 'primary-button-dark'
+            }
+        }
+
+        const getDangerModeVariantClassName = (variant: ButtonVariant) => {
+            switch (variant) {
+                case 'primary':
+                    return "usa-button--secondary"
+                case "secondary":
+                    return "usa-button--outline outline-danger"
+                case 'link':
+                    return "usa-button--unstyled unstyled-button-danger"
                 default:
                     return getVariantClassName(variant)
             }
         }
 
-        return this.mode === "light"
-            ? `usa-button ${getVariantClassName(this.variant)}`
-            : `usa-button ${getDarkModeVariantClassName(this.variant)}`
+        let getClassName;
+        switch (this.mode) {
+            case "light":
+                getClassName = getVariantClassName;
+                break;
+            case "dark":
+                getClassName = getDarkModeVariantClassName;
+                break;
+            case "danger":
+                getClassName = getDangerModeVariantClassName;
+                break;
+            default:
+                getClassName = getVariantClassName;
+        }
+
+        return `usa-button ${getClassName(this.variant)} ${this.icon ? 'button-icon' : ''}`
     }
 
-    componentWillLoad() {
-        if (this.asChild) {
-            const slotElements = this.hostElement.children
-            if (slotElements.length !== 1) {
-                throw new Error(`If the asChild property is set to true on the njwds-button component, the component must have exactly one slot element. Instead got ${slotElements.length} elements.`)
-            }
-            if (slotElements[0].tagName !== "BUTTON") {
-                throw new Error(`If the asChild property is set to true on the njwds-button component, the slot element must be a <button>. Instead got ${slotElements[0].outerHTML}`)
-            }
-            const buttonElement = slotElements[0] as HTMLButtonElement
-            const buttonClassName = this.getButtonClassName()
-            buttonElement.className = `${buttonElement.className} ${buttonClassName}`
+    private renderIcon() {
+        let iconClass = ''
+        switch (this.iconPosition) {
+            case "leading":
+                iconClass += ' margin-right-105'
+                break;
+            case "trailing":
+                iconClass += ' margin-left-105'
+                break;
         }
 
-    }
+        if (this.icon) {
+            if (this.iconPosition === 'icon-only') {
+                return (
+                  <njwds-icon icon={this.icon} size="3" decorative={false} iconTitle={this.iconTitle}></njwds-icon>
+                )
 
+            } else {
+                return (
+                  <njwds-icon class={iconClass} icon={this.icon} size="3" decorative={true} iconTitle={this.iconTitle}></njwds-icon>
+                )
+            }
```

Comment:
[Dust] I'm wondering if it'd be clearer if we just had a single return statement and we conditionally change the attribute for `decorative` and `class` so we don't need to repeat the rest of the attributes for both. It may be clearer to the dev which things are static and which are changing